packages/inquirer/src/ui/prompt.mts

```diff
@@ -37,6 +37,10 @@ const _ = {
         pointer[key] = {};
       }
 
+      if (index !== arr.length - 1 && typeof pointer[key] === 'boolean') {
```

Comment:
Do we need to make a specific case for boolean? Sounds like it's just the behavior we should get no matter what 🤔 - Even wonder if the issue is the condition on line 36 `!(key in pointer)` - maybe that line should just always run. Or typecheck if not object.

Reply:
That's actually a good call. I didn't realize at first glance that trying to override a string value would also cause this error.

Updated to do an object check instead.

Reply:
(side note, I looked at running line 37 every time and it blew up other tests. 😅)

Reply:
Uh weird, works for me - I pushed the changes.